### PR TITLE
MM-46893 :  e2e for leave and archive channel destructive

### PR DESCRIPTION
**Pull request by AbhinavVihan**
#### Summary
e2e for leave and archive channel destructive

#### Ticket Link
Fixes: https://github.com/mattermost/mattermost-server/issues/20951
JIRA: https://mattermost.atlassian.net/browse/MM-46893

#### Related Pull Requests
None

#### Screenshots
None

#### Release Note
None

```release-note
NONE
```

### CODE REVIEW

**Review by yasserfaraazkhan — APPROVED** [review]
💯 Looks great!!

@AbhinavVihan small nit: Please remove the double `//` for comment lines

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+        // // * Archive Channel menu option should be visible and hav a background-color (destructive);
+        cy.get('#channelArchiveChannel').should('be.visible').children().focus().should('have.css', 'background-color', 'rgb(210, 75, 78)');
+
+        // // * Leave Channel menu option should be visible and hav a background-color (destructive);
+        cy.get('#channelLeaveChannel').should('be.visible').children().focus().should('have.css', 'background-color', 'rgb(210, 75, 78)');
+    });
+
+    it('MM-T4943_2 Leave channel actions display as destructive in the Edit Channel Menu ', () => {
+        // # Open Edit Channel Menu and verify menu optins
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:75]:
```suggestion
        // # Open Edit Channel Menu and verify menu option
```

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+            cy.apiLogin(testUser);
+            cy.visit(offTopicUrl);
+        });
+    });
+
+    it('MM-T4943_1 Leave and Archive channel actions display as destructive in the channel dropdown menu', () => {
+        // # click on channel drop-down menu
+        cy.get('#channelHeaderTitle').should('be.visible').click();
+
+        // // * View Info menu option should be visible;
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:34]:
```suggestion
        // * View Info menu option should be visible;
```

Can you remove double // , from rest of the comments

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        // // * Edit Channel Header menu option should be visible;
+        cy.get('#channelEditHeader').should('be.visible');
+
+        // // * Edit Channel Purpose menu option should be visible;
+        cy.get('#channelEditPurpose').should('be.visible');
+
+        // // * Rename Channel menu option should be visible;
+        cy.get('#channelRename').should('be.visible');
+
+        // // * Archive Channel menu option should be visible and hav a background-color (destructive);
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:67]:
```suggestion
        // // * Archive Channel menu option should be visible and have a background-color (destructive);
```

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        // // * New Category Sub-menu option should be visible;
+        cy.findByText('New Category').should('be.visible');
+
+        // // * Copy Link menu option should be visible;
+        cy.findByText('Copy Link').should('be.visible');
+
+        // // * Add Members menu option should be visible;
+        cy.findByText('Add Members').should('be.visible');
+
+        // // * Leave Channel menu option should be visible and hav a color (destructive);
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:99]:
```suggestion
        // // * Leave Channel menu option should be visible and have a color (destructive);
```

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        cy.findByText('Leave Channel').should('be.visible').should('have.css', 'color', 'rgb(210, 75, 78)');
+
+        // // * Leave Channel menu option should be visible and hav a background-color (destructive);
```
Comment (furqanmlk) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:102]:
```suggestion
        // // * Leave Channel menu option should be visible and have a background-color (destructive);
```

**Review by furqanmlk — COMMENTED** [review]
Thanks @AbhinavVihan 
Looks good to me. Just few minor changes

**Review by furqanmlk — APPROVED** [review]
Thanks @AbhinavVihan

File: e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+        // * Archive Channel menu option should be visible and have a background-color (destructive)
+        cy.get('#channelArchiveChannel').should('be.visible').children().focus().should('have.css', 'background-color', 'rgb(210, 75, 78)');
+
+        // * Leave Channel menu option should be visible and hav a background-color (destructive)
+        cy.get('#channelLeaveChannel').should('be.visible').children().focus().should('have.css', 'background-color', 'rgb(210, 75, 78)');
+    });
+
+    it('MM-T4943_2 Leave channel actions display as destructive in the Edit Channel Menu ', () => {
+        // # Open Edit Channel Menu and verify menu optin
+        cy.get('#sidebarItem_off-topic').find('.SidebarMenu_menuButton').click({force: true});
```
Comment (Timoteo714) [inline e2e/cypress/tests/integration/channel/leave_and_archive_channel_destructive_spec.ts:76]: Mobile

**Review by Timoteo714 — COMMENTED** [review]
image

**Pull request merged**